File: src/io/innofang/Decorator/example/FruitCake.java
package io.innofang.Decorator.example;

/**
 * Created by Inno Fang on 2017/5/4.
 */
public class FruitCake extends DecorateCake {

    public FruitCake(Cake cake) {
        super(cake);
    }

    @Override
    public void make() {
        addSomeFruit();
        super.make();
    }

    private void addSomeFruit() {
        System.out.println("Add some fruit");
    }
}

File: src/io/innofang/decorator/example/kotlin/FruitCake.kt
package io.innofang.decorator.example.kotlin

/**
 * Created by Inno Fang on 2017/9/4.
 */
class FruitCake
constructor(cake: Cake) : DecorateCake(cake) {
    override fun make() {
        addSomeFruit()
        super.make()
    }

    private fun addSomeFruit() {
        println("Add some fruit")
    }
}